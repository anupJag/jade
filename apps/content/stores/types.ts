export type ContentAppLabels = {
  [key: string]: string;
};

export type ContentAppConfig = {
  [key: string]: string;
};

export type State = {
  label: ContentAppLabels;
  config: ContentAppConfig;
};

export type Action = { type: string };
