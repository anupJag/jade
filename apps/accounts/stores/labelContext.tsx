import React from 'react';

export interface LabelContext {}

export const LabelContext = React.createContext(null);

export default LabelContext;
