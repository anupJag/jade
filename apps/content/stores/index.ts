export { ContentContext, ContentContextProvider } from './Context';
