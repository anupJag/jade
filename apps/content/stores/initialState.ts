import { State } from './types';

export const initialState: State = {
  label: {},
  config: {},
};
