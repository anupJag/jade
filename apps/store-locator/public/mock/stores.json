[
  {
    "id": 1001,
    "name": "Charing Cross Express",
    "address": "1-4 Charing Cross, London",
    "country": "UK",
    "postal_code": "SW1A 2DR",
    "contact": "0345 026 9353",
    "location": {
      "latitude": 51.507404,
      "longitude": -0.12724
    },
    "open_hours": {
      "mon": "7am - 11pm",
      "tue": "7am - 11pm",
      "wed": "7am - 11pm",
      "thu": "7am - 11pm",
      "fri": "7am - 11pm",
      "sat": "7am - 11pm",
      "sun": "7am - 11pm"
    },
    "openHourInfo": "Opening hours vary for some facilities. View store details for more information.",
    "isOpen": true
  },
  {
    "id": 1002,
    "name": "Covent Garden Metro",
    "address": "22-25 Bedford St, Covent Garden, London",
    "country": "UK",
    "postal_code": "WC2E 9EQ",
    "contact": "0345 026 9353",
    "location": {
      "latitude": 51.507404,
      "longitude": -0.12724
    },
    "open_hours": {
      "mon": "7am - 11pm",
      "tue": "7am - 11pm",
      "wed": "7am - 11pm",
      "thu": "7am - 11pm",
      "fri": "7am - 11pm",
      "sat": "7am - 11pm",
      "sun": "7am - 11pm"
    },
    "openHourInfo": "Opening hours vary for some facilities. View store details for more information.",
    "isOpen": false
  },
  {
    "id": 1003,
    "name": "Warrington Man Rd Express",
    "address": "Manchester Rd, Warrington",
    "country": "UK",
    "postal_code": "WA1 3NJ",
    "contact": "0345 026 9353",
    "location": {
      "latitude": 53.397055,
      "longitude": -2.565406
    },
    "open_hours": {
      "mon": "7am - 11pm",
      "tue": "7am - 11pm",
      "wed": "7am - 11pm",
      "thu": "7am - 11pm",
      "fri": "7am - 11pm",
      "sat": "7am - 11pm",
      "sun": "7am - 11pm"
    },
    "openHourInfo": "Opening hours vary for some facilities. View store details for more information.",
    "isOpen": true
  },
  {
    "id": 1004,
    "name": "Warrington Orford Green Esso Express",
    "address": "Orford Green, Warrington, Warrington",
    "country": "UK",
    "postal_code": "WA2 8PB",
    "contact": "0345 026 9353",
    "location": {
      "latitude": 53.406646,
      "longitude": -2.57579
    },
    "open_hours": {
      "mon": "7am - 11pm",
      "tue": "7am - 11pm",
      "wed": "7am - 11pm",
      "thu": "7am - 11pm",
      "fri": "7am - 11pm",
      "sat": "7am - 11pm",
      "sun": "7am - 11pm"
    },
    "openHourInfo": "Opening hours vary for some facilities. View store details for more information.",
    "isOpen": true
  },
  {
    "id": 1005,
    "name": "Dam Lane Express",
    "address": "52-54 Dam Lane, Warrington",
    "country": "UK",
    "postal_code": "WA1 4EP",
    "contact": "0345 026 9353",
    "location": {
      "latitude": 53.401793,
      "longitude": -2.531151
    },
    "open_hours": {
      "mon": "7am - 11pm",
      "tue": "7am - 11pm",
      "wed": "7am - 11pm",
      "thu": "7am - 11pm",
      "fri": "7am - 11pm",
      "sat": "7am - 11pm",
      "sun": "7am - 11pm"
    },
    "openHourInfo": "Opening hours vary for some facilities. View store details for more information.",
    "isOpen": false
  }
]
