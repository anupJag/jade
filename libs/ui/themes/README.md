# fe-themes

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test fe-themes` to execute the unit tests via [Jest](https://jestjs.io).
