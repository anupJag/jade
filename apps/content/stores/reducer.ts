import { State, Action } from './types';
import { initialState } from './initialState';

const contentReducers = (state: State = initialState, action: Action = { type: '' }) => {
  return state;
};

export default contentReducers;
