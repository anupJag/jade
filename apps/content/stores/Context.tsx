/** @context.tsx
 *  This component is a wrapper which will provide
 *  the Global Context to other needy components
 *  */

import React, { useReducer } from 'react';
import { initialState } from './initialState';
import reducer from './reducer';
import { Action, State } from './types';

interface ContentContextProps {
  state: State;
  dispatch: React.Dispatch<Action>;
}

export const ContentContext = React.createContext<ContentContextProps>({
  state: initialState,
} as ContentContextProps);

// eslint-disable-next-line react/prop-types
export const ContentContextProvider = ({ children, ssrAppState = {} }) => {
  const [state, dispatch] = useReducer(reducer, { ...initialState, ...ssrAppState });

  return <ContentContext.Provider value={{ state, dispatch }}>{children}</ContentContext.Provider>;
};
