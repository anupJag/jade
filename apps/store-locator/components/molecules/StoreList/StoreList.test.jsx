import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@jade/ui/themes';

import { StoreList } from './index';
import { LabelContext } from '../../../stores/LabelContext';

describe('StoreList', () => {
  let StoreListComponent = '';

  beforeEach(() => {
    const data = [
      {
        id: '5f93f0d2d64785a0f430a024',
        name: 'Covent Garden Metro',
        description: 'Covent Garden Metro',
        address: {
          addressLine1: '22-25 Bedford St,',
          addressLine2: 'Covent Garden, London',
          city: 'London',
          state: 'London',
          country: 'United Kingdom',
          postalCode: 'WC2E 9EQ',
        },
        contactNo: '0345 026 9353',
        location: {
          latitude: 51.507404,
          longitude: -0.12724,
        },
        openHourInfo: 'Opening hours vary for some facilities.',
        thumbnailImage:
          'https://images.ctfassets.net/r1nemv2qs9l3/6jfs8jFS2U8346IVZSKVMs/0e9ba48db1fc04c39ea644c97d6c57aa/thumb1.jpg?fit=fill&fm=jpg&fl=progressive&h=200&w=200',
        images: [
          'https://images.ctfassets.net/r1nemv2qs9l3/2srrAwkz57zwaSCL5IRDTs/7103a35e889b2dd2339fc3ad9f8395d9/store1.jpg?fit=fill&fm=jpg&fl=progressive&h=720&w=1280',
          'https://images.ctfassets.net/r1nemv2qs9l3/2ODKVNobEPcoJFNaFGkPia/8f9c93f610091fbaec6b54a4b03c88f3/store2.jpeg?fit=fill&fm=jpg&fl=progressive&h=720&w=1280',
        ],
        openHours: {
          mon: {
            openingTime: '07:00',
            closingTime: '23:00',
          },
          tue: {
            openingTime: '07:00',
            closingTime: '23:00',
          },
          wed: {
            openingTime: '07:00',
            closingTime: '23:00',
          },
          thu: {
            openingTime: '07:00',
            closingTime: '23:00',
          },
          fri: {
            openingTime: '07:00',
            closingTime: '23:00',
          },
          sat: {
            openingTime: '07:00',
            closingTime: '23:00',
          },
          sun: {
            openingTime: '07:00',
            closingTime: '23:00',
          },
        },
        timezone: '+5.0',
      },
      {
        id: '5f9a2da8e887a1f7ce0fa752',
        name: 'Charing Cross Express',
        description: 'Covent Garden Metro',
        address: {
          addressLine1: '1-4 Charing Cross',
          addressLine2: '1-4 Charing Cross',
          city: 'London',
          state: 'London',
          country: 'United Kingdom',
          postalCode: 'SW1A 2DR',
        },
        contactNo: '0345 026 9353',
        location: {
          latitude: 51.507404,
          longitude: -0.12734,
        },
        openHourInfo: 'Opening hours vary for some facilities.',
        thumbnailImage:
          'https://images.ctfassets.net/r1nemv2qs9l3/6jfs8jFS2U8346IVZSKVMs/0e9ba48db1fc04c39ea644c97d6c57aa/thumb1.jpg?fit=fill&fm=jpg&fl=progressive&h=200&w=200',
        images: [
          'https://images.ctfassets.net/r1nemv2qs9l3/2srrAwkz57zwaSCL5IRDTs/7103a35e889b2dd2339fc3ad9f8395d9/store1.jpg?fit=fill&fm=jpg&fl=progressive&h=720&w=1280',
          'https://images.ctfassets.net/r1nemv2qs9l3/2ODKVNobEPcoJFNaFGkPia/8f9c93f610091fbaec6b54a4b03c88f3/store2.jpeg?fit=fill&fm=jpg&fl=progressive&h=720&w=1280',
        ],
        openHours: {
          mon: '7am - 11pm',
          tue: '7am - 11pm',
          wed: '7am - 11pm',
          thu: '7am - 11pm',
          fri: '7am - 11pm',
          sat: '7am - 11pm',
          sun: '7am - 11pm',
        },
        timezone: '+5.0',
      },
    ];

    const defaultLabels = {
      slResultsHeading: 'Store locator',
      slResultsSubHeading1: 'Showing nearest',
      slResultsSubHeading2: 'resuts for',
      slStoreDetailBtnLabel: 'View store details',
    };

    StoreListComponent = render(
      <ThemeProvider>
        <LabelContext.Provider value={defaultLabels}>
          {/* <StoreList data={data}/> */}
        </LabelContext.Provider>
      </ThemeProvider>,
    );
  });

  it('should render store basic details correctly', () => {
    const { asFragment } = StoreListComponent;
    expect(asFragment()).toMatchSnapshot();
  });
});
