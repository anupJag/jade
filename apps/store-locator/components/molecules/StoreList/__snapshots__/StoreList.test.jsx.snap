// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreList should render store basic details correctly 1`] = `<DocumentFragment />`;
